R/mobr.R

```diff
@@ -499,392 +797,156 @@ samp_ssad = function(comm, groups){
-              min_plot_group = min(group_plots$Freq[which(as.character(group_plots[, 1]) %in% 
-                                                            c(as.character(group), as.character(ref_group)))])
-              ddeltaS_group = sample_rare_keep$deltaS[sample_rare_keep$group == group][1:min_plot_group] - 
-                sample_rare_keep$deltaS[as.character(sample_rare_keep$group) == as.character(ref_group)][1:min_plot_group]
-              
-              group_for_2 = env_data[which(as.character(env_data) %in% c(as.character(ref_group), as.character(group)))]
-              comm_group = comm$comm[as.character(env_data) == as.character(group), ]
-              impl_S_group = sample_rare_keep$impl_S[which(as.character(sample_rare_keep$group) == as.character(group))]
-              impl_S_group = as.numeric(as.character(impl_S_group))
-              
-              null_agg_deltaS_mat = matrix(NA, nperm, min_plot_group)
-              cat('\nComputing null model for aggregation effect\n')
-              pb <- txtProgressBar(min = 0, max = nperm, style = 3)
-              for (i in 1:nperm){
-                setTxtProgressBar(pb, i)
-                xy_perm = comm$spat[sample(nrow(comm$spat)), ]
-                xy_perm_group = xy_perm[as.character(env_data) == as.character(group), ]
-                xy_perm_ref = xy_perm[as.character(env_data) == as.character(ref_group), ]
-                expl_S_perm_group = rarefy_sample_explicit(comm_group, xy_perm_group)
-                expl_S_perm_ref = rarefy_sample_explicit(ref_comm, xy_perm_ref)
-                null_agg_deltaS_mat[i, ] = expl_S_perm_group[1:min_plot_group] - impl_S_group[1:min_plot_group] - 
-                  (expl_S_perm_ref[1:min_plot_group] - impl_S_ref[1:min_plot_group])
-              }
-              close(pb)
-              agg_deltaS_null_CI = apply(null_agg_deltaS_mat, 2, function(x) quantile(x, c(0.025, 0.5, 0.975)))
-              agg_group = data.frame(cbind(rep(as.character(group), min_plot_group),1:min_plot_group,  
-                                           ddeltaS_group, t(agg_deltaS_null_CI)))
-              out$discrete$agg = rbind(out$discrete$agg, agg_group)
-            }
-          }
-          names(out$discrete$agg) = c('group', 'effort_sample', 'ddeltaS_emp', 'ddeltaS_null_low', 
-                                      'ddeltaS_null_median', 'ddeltaS_null_high')
-        }
-      }
+    if (type == 'continuous'){
+        get_delta_continuous_checks(corr, group_levels, env_levels)
+        if ('SAD' %in% approved_tests)
+            out = effect_SAD_continuous(out, group_sad, env_levels, nperm)
+        if ('N' %in% approved_tests)
+            out = effect_N_continuous(out, comm, S, group_levels, env_levels, 
+                                      group_data, plot_dens, plot_abd, 
+                                      ind_sample_size, nperm)
+        # Effect of aggregation
+    } else {
```

Comment:
rather than `else` we should make the function break if type does not equal `discrete` just in case the user supplies some strange argument for `type`. Change this line to `if (type == 'discrete') {`